### DISCUSSION TITLE
Problem when gdb the programme which use c++ with swig as a module and python as the interface(Linux)

### DISCUSSION THREAD

**Discussion by David-JA** [swig/swig#2248] [General]
Swig is a excellent projet and our team has benefit a lot from that, but both of us has a question about how to debug the python file and  the c++ source code at the same time. The fact is that the gdb can't catch the symbol of the .os dynamic link library created by swig, which result in the debug problem in vscode.
When we try the example as the other people who develop a C extension for python and then debug the extension, the gdb can catch the symbol, like the example in this question https://stackoverflow.com/questions/66077410/debug-a-python-c-c-extension-in-vscode-on-windows. But it's not the same for us to debug our programme with swig, even with the compiler option -O0 -g3.
Could you please help us? It's there any other exemple for debug the c++/python code with vscode in the same time?

**Comment by ojwb** [comment]
This is a usage question rather than a bug with SWIG so I'm converting to a discussion.

You may have more luck asking on the swig-user mailing list though.

**Reply by David-JA** [reply]
Thanks you! We find one of the problem is about the limitation of cppdbg, which is the debugger extensions of microsoft for VS CODE, it has the special token for two debugger to attach the same process at the same time, and to catch the debug symbol.
This it's why we can't catch the debug symbol for both pdb and gdb on the same process, we use the other gdb extension and assign the PID manually, it‘s ok, although it's needs the user to change the debugger between c++ and python manually with some other disadvantages, but it works.
Holp the other user who has the same requirement can benefit from our experience!

**Reply by ojwb** [reply]
Seems you've solved the problem, so closing.
